### ISSUE TITLE
Trilinos: New ATS2 environment is modifying a system level ENV variable that breaks Spectrum

### ISSUE THREAD

**Issue by jjellio** [trilinos/Trilinos#6875]
## Bug Report
@trilinos/framework (I think?)

### Description
In the ATS2 `environment.sh` the `TMPDIR` variable is modified. This causes problems with Spectrum (and perhaps other system level tools) that depend on `TMPDIR`
```
atdm/ats2/environment.sh:174:  export TMPDIR=/tmp/$(whoami)
atdm/ats2/environment.sh:175:  if [[ ! -d ${TMPDIR} ]] ; then
atdm/ats2/environment.sh:176:    echo "Creating ${TMPDIR} for nvcc wrapper!"
atdm/ats2/environment.sh:177:    mkdir ${TMPDIR}
```

### Steps to Reproduce
1. load the ats2 environment on Vortex
2. jsrun will fail after

@e10harvey @bartlettroscoe 

I will be submitting a PR that fixes this

**Comment by jjellio** [comment]
I found the use of TMPDIR in `nvcc_wrapper`
```
./packages/kokkos/bin/nvcc_wrapper:86:temp_dir=${TMPDIR:-/tmp}
```

It may be best to simply delete all of the `TMPDIR` stuff from the `environment.sh`, and let Kokkos default to `/tmp`. I'm guessing this functionality was added to support multiple users sharing a node. If the testing tools want to communicate a different TMP to Kokkos, then Kokkos and the Testing group need to figure out a different variable name E.g., `NVCC_WRAPPER_TMPDIR` (which could default to `TMPDIR`.

@trilinos/kokkos

**Comment by e10harvey** [comment]
@jjellio: Good catch. Thank you for looking into this! The nvcc_wrapper you've listed above will default to ${TMPDIR} on systems that have TMPDIR set at login. Logging into vortex shows that TMPDIR is set. Does the entity writing to `/tmp/$(whoami)` (per 
23eff8e) need to be writing to this `temp_dir=${TMPDIR:-/tmp}` location?

**Comment by bartlettroscoe** [comment]
@jjellio, to be clear, does this fix #6861?

@e10harvey, just curious, but where did you get the code from the set:

```
export TMPDIR=/tmp/$(whoami)
```

?

I added the code to create that directory if it was missing but the statement `export TMPDIR=/tmp/$(whoami)` was already there.

**Comment by e10harvey** [comment]
> @e10harvey, just curious, but where did you get the code from the set:
> 
> ```
> export TMPDIR=/tmp/$(whoami)
> ```

I wrote it. I didn't realize TMPDIR was a reserved env var. My mistake.

**Comment by bartlettroscoe** [comment]
> I wrote it. I didn't realize TMPDIR was a reserved env var. My mistake.

@e10harvey, what was the motivation for setting `TMPDIR`?  Do you remember?  Just curious.

**Comment by e10harvey** [comment]
> @e10harvey, what was the motivation for setting `TMPDIR`? Do you remember? Just curious.

Something in the build system was trying to access `/tmp/$(whoami)` which did not exist.

**Comment by bartlettroscoe** [comment]
> Something in the build system was trying to access `/tmp/$(whoami)` which did not exist.

@e10harvey, right, that is why I added the code to create that directory but that does not explain why `export TMPDIR=/tmp/$(whoami)` got set.

@jjellio and @e10harvey, do we need to set:

```
export TMPDIR=/tmp/$(whoami)
```

while building with `nvcc` and then set it to:

```
export TMPDIR=/tmp
```

when running the tests with jsrun?

**Comment by e10harvey** [comment]
> @jjellio and @e10harvey, do we need to set:

@bartlettroscoe: TMPDIR is set in the vortex env. Apparently LSF uses it on vortex and we are not supposed to overwrite it (see description in this PR). Something has been trying to access `/tmp/$(whoami)` so, I haven't tested the changes in this PR, but I think that @jjellio's fix in 
23eff8e is all that we need. The one thing I'm not sure about is whether the entity that is accessing `/tmp/$(whoami)` needs to be accessing the same directory that `nvcc_wrapper` is -- I suspect that answer to that is **no** since `nvcc_wrapper` defaults to tmp happily and the entity (compiler?) defaults to `/tmp/$(whoami)` happily. Why don't we get this merged and let the nightly jenkins jobs test this?

**Comment by bartlettroscoe** [comment]
@e10harvey, perhaps we should just change this to be:

```
if [[ ! -d /tmp/${USER} ]]; then
  echo "Creating missing /tmp/${USER}"
  mkdir /tmp/${USER}
fi
```

an not **NOT** set `TMPDIR`.  There is no harm in just creating `/tmp/${USER}`.  Does that get around build problems with `nvcc`?

**Comment by e10harvey** [comment]
>  Does that get around build problems with `nvcc`?


Let's merge this and see.

**Comment by jjellio** [comment]
If you look at the SOURCE for NVCC_WRAPPER, it wants a `tmpdir`, which it sets from `TMPDIR`

IMO, Kokkos needs to not use TMPDIR directly, and instead document a namespaced (prefixed) ENV for NVCC_WRAPPER, e.g., `NVCC_WRAPPER_TMPDIR`, that way if the testing infrastructure needs to configure this they can do so safely.  Then, Kokkos' sets `NVCC_WRAPPER_TMPDIR` to default to `TMPDIR/whoami`

That removes the need to change TMPDIR, and preserved functionality for downstream user to change it.

**Comment by jjellio** [comment]
@bartlettroscoe My patch fixed things on Vortex/Rzansel (lassen was unusually functional without it).

Once this goes through (and hopefully Kokkos/Framework can think about my prior comment). Then we can see how your RM connection issue behaves.  There are still likely issues there, but they will be independent of this.

**Comment by bartlettroscoe** [comment]
FYI: With the Trilinos PR testing system down with no ETA to fix, I manually merged the branch in PR #6876 to the 'atdm-nightly-manual-updates' branch just now in commit 47b673b so it will be in the 'atdm-nightly' branch tonight and we will see this running tomorrow.

Putting this Issue in review to see if this fixes the problem.
